File: bowling.py
def calculate_score(bowls):
    score = 0
    frame = 1
    i = 0
    
    def get_frame_score(bowls, i):
        score1 = 0 if bowls[i] == '-' else int(bowls[i])
        score2 = 0 if bowls[i+1] == '-' else int(bowls[i+1])
        return score1 + score2
    
    def get_strike_bonus(bowls, i):
        bonus = 0
        if bowls[i+2] == 'X':
            bonus += 10
        elif bowls[i+2] == '/':
            bonus += 10 - int(bowls[i+1])
        else:
            bonus += int(bowls[i+1]) + int(bowls[i+2])

        return bonus

    def get_spare_bonus(bowls, i):
        bonus = 0
        if bowls[i+2] == 'X':
            bonus += 10
        else:
            bonus += int(bowls[i+2])

        return bonus
    
    while frame <= 10:
        if bowls[i] == 'X':
            score += 10
            score += get_strike_bonus(bowls, i)
            i += 1
        elif bowls[i+1] == '/':
            score += 10
            score += get_spare_bonus(bowls, i)
            i += 2
        else:
            score += get_frame_score(bowls, i)
            i += 2
        
        frame += 1
    
    return score

bowls = input()
print(calculate_score(bowls))

File: bowling.cpp
#include <iostream>
#include <string>

int getFrameScore(char bowl1, char bowl2) {
    int score = 0;
    if (bowl1 == 'X') {
        score = 10;
    } else if (bowl1 == '/') {
        score = 10 - (bowl2 - '0');
    } else {
        score = (bowl1 - '0') + (bowl2 - '0');
    }
    return score;
}

int getScore(std::string input) {
    int score = 0;
    int frame = 0;
    for (int i = 0; i < input.length(); i++) {
        if (frame == 10) {
            break;
        }
        if (input[i] == 'X') {
            score += 10 + getFrameScore(input[i+1], input[i+2]);
            frame++;
            if (frame < 10) {
                i++;
            }
        } else if (input[i] == '/') {
            score += 10 - (input[i-1] - '0') + getFrameScore(input[i+1], input[i+2]);
            frame++;
            if (frame < 10) {
                i++;
            }
        } else {
            score += (input[i] - '0');
            if (input[i+1] != '-') {
                score += (input[i+1] - '0');
            }
            frame++;
            if (frame < 10) {
                i++;
            }
        }
    }
    return score;
}

int main() {
    std::string input;
    std::cin >> input;
    int score = getScore(input);
    std::cout << score << std::endl;
    return 0;
}